backend/src/paymaster/pimlico.test.ts

```diff
@@ -0,0 +1,324 @@
+import { providers, ethers } from "ethers";
+import { PimlicoPaymaster, getERC20Paymaster } from "./pimlico.js";
+import { NATIVE_ASSET, ORACLE_ADDRESS, TOKEN_ADDRESS } from "../../src/constants/Pimlico.js";
+
+describe('PimlicoPaymaster on Mumbai', () => {
+  const paymasterAddress = '0x32aCDFeA07a614E52403d2c1feB747aa8079A353'; // Mumbai Etherspot Paymaster Address
+  const entryPointAddress = "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789"; // EntryPoint v0.6 as default
+  const bundlerUrl = 'https://mumbai-bundler.etherspot.io';
+  const userOp = {
+    sender: '0x7b3078b9A28DF76453CDfD2bA5E75f32f0676321',
+    nonce: '0x1',
+    initCode: '0x',
+    callData: '0x47e1da2a000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000a000000000000000000000000000000000000000000000000000000000000000e0000000000000000000000000000000000000000000000000000000000000000100000000000000000000000080a1874e1046b1cc5defdf4d3153838b72ff94ac0000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000002386f26fc10000000000000000000000000000000000000000000000000000000000000000000100000000000000000000000000000000000000000000000000000000000000200000000000000000000000000000000000000000000000000000000000000000',
+    callGasLimit: '0x88b8',
+    verificationGasLimit: '0x186a0',
+    maxFeePerGas: '0x6fc23ac10',
+    maxPriorityFeePerGas: '0x6fc23ac00',
+    paymasterAndData: '0x0101010101010101010101010101010101010101000000000000000000000000000000000000000000000000000001010101010100000000000000000000000000000000000000000000000000000000000000000101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101',
+    signature: '0x0101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101010101',
+    preVerificationGas: '0xc6c4'
+  };
+
+  const ERC20PaymasterBuildOptions = {
+    entrypoint: "0x5FF137D4b0FDCD49DcA30c7CF57E578a026d2789",
+    nativeAsset: NATIVE_ASSET[80001],
+    nativeAssetOracle: ORACLE_ADDRESS[80001].MATIC,
+    tokenAddress: TOKEN_ADDRESS[80001].USDC,
+    tokenOracle: ORACLE_ADDRESS[80001].USDC,
+    owner: "0x4337000c2828f5260d8921fd25829f606b9e8680",
+    deployer: "0x4337000c2828f5260d8921fd25829f606b9e8680"
+  }
+
+  const provider = new providers.JsonRpcProvider(bundlerUrl);
+  const pimlicoPaymaster = new PimlicoPaymaster(paymasterAddress, provider);
+
+  test('calculateTokenAmount function - valid details', async () => {
+    try {
+      await pimlicoPaymaster.calculateTokenAmount(userOp);
+    } catch (e) {
+      throw new Error('An error is displayed while performing calculateTokenAmount action.')
+    }
+  });
+
+  test('generatePaymasterAndData function - valid details', async () => {
+    try {
+      await pimlicoPaymaster.generatePaymasterAndData(userOp);
```

Comment:
I don't see any assertion here? How is this supposed to work? What am I missing?